x/leverage/keeper/keeper.go

```diff
@@ -123,3 +125,157 @@ func (k Keeper) WithdrawAsset(ctx sdk.Context, lenderAddr sdk.AccAddress, uToken
 
 	return nil
 }
+
+// SetCollateral enables or disables a uToken denom for use as collateral by a single borrower.
+func (k Keeper) SetCollateral(ctx sdk.Context, borrowerAddr sdk.AccAddress, denom string, enable bool) error {
+	if !k.IsAcceptedUToken(ctx, denom) {
+		return sdkerrors.Wrap(types.ErrInvalidAsset, denom)
+	}
+
+	// TODO: Enable sets to true; disable removes from KVstore rather than setting false
+
+	return nil
+}
+
+// GetLoan returns an sdk.Coin representing how much of a given denom a borrower currently owes.
+func (k Keeper) GetLoan(ctx sdk.Context, borrowerAddr sdk.AccAddress, denom string) sdk.Coin {
+	store := ctx.KVStore(k.storeKey)
+	owed := sdk.NewCoin(denom, sdk.ZeroInt())
+	key := types.CreateLoanKey(borrowerAddr, denom)
+	if store.Has(key) {
+		err := owed.Amount.Unmarshal(store.Get(key))
+		if err != nil {
+			// Should never happen
+			panic(err)
+		}
+	}
+	return owed
+}
+
+// GetAllLoans returns an sdk.Coins object containing all open borrows associated with an address
+func (k Keeper) GetAllLoans(ctx sdk.Context, borrowerAddr sdk.AccAddress) (sdk.Coins, error) {
+	currentlyBorrowed := sdk.NewCoins()
+	store := ctx.KVStore(k.storeKey)
+	var key []byte
+	key = append(key, types.KeyPrefixLoanToken...)
+	key = append(key, address.MustLengthPrefix(borrowerAddr)...)
+	prefixStore := prefix.NewStore(store, key)
+	iter := prefixStore.Iterator(nil, nil) // TODO: Find out why iter starts with valid = false
+	//	Alternative:
+	//  iter := sdk.KVStorePrefixIterator(store, key) // TODO: Same problem
```

Comment:
Note: This is the place where iterator is invalid when created. It might also be a problem with the prefixStore or the key.

Reply:
Note to self: it was a misunderatanding on my part - it's _supposed_ to be invalid once all prefixed keys have been iterated through (including when created and no qualifying keys exist).